Wiki Rewrite: "How To Install"

its-miroma opened the pull request:
### Preview: <https://its-miroma.gitbook.io/fo/how-to/install>

- [ ] Do we want to use [GitBook's Stepper blocks](https://docs.gitbook.com/content-editor/blocks/stepper)?

<details>
<summary>Summary of the Plan</summary>

- The current content will remain as-is until all the new pages are merged.
- The new content will be hidden until everything is merged.
- The last PR will:
  - Show all new content
  - Delete old files
  - Symlink old files through `.gitbook.yaml` to avoid breaking links
    - I will still try to update links in other FO repos
  - Update `README`

</details>

<details>
<summary>Formatting</summary>

|     Format       | When         | Example                             |
| :--------------: | ------------ | ----------------------------------- |
|      `code`      | versions     | Select version `1.0.0`              |
|      `code`      | paths        | Open `~/.minecraft/`                |
|      "text"      | to be typed  | Search for "Fabulously Optimized"   |
|   <kbd>A</kbd>   | keybinds     | Press <kbd>↵ Enter</kbd> twice      |
| <kbd>**A**</kbd> | UI buttons   | Click on <kbd>**Download**</kbd>    |
|  <kbd>`A`</kbd>  | UI values    | Set the value to <kbd>`True`</kbd>  |
|  _`italiccode`_  | steps, lines | Follow _`step 1`_ for _`line 20`_   |

</details>

its-miroma committed:
rewrite(how-to/install): initial commit

its-miroma committed:
rewrite(how-to/install): fix url

its-miroma committed:
rewrite(how-to/install): fix headings

Madis0 commented on how-to/install/README.md:
```diff
@@ -0,0 +1,20 @@
+{% hint style="success" %}
+FO does not require any special setup server-side: any server compatible with vanilla should be supported.
+
+Read more [tips regarding server setup](../../about/servers.md).
```
possibly broken link

Madis0 commented on how-to/install/vanilla.md:
```diff
@@ -0,0 +1,197 @@
+{% tab title="Windows" %}
+1. Download the [FO Installer](https://download.fo/vanilla)
+2. Open the `exe` installer. If you get a security prompt from Windows, allow the installer to run:
+   * [Why did I get this prompt?](#why-does-windows-show-the-security-prompt)
```
could just have an inline "(why?)" or at least have it as the last bullet point; as first it is confusing

Madis0 commented on how-to/install/vanilla.md:
```diff
@@ -0,0 +1,197 @@
+{% tab title="macOS 15 Sequoia or later" %}
+1. Download the [FO Installer](https://download.fo/vanilla)
+2. Open the `dmg` installer. If you get a security prompt from macOS, allow the installer to run:
+   * [Why did I get this prompt?](#why-does-macos-show-the-security-prompt)
```
see comment above

Madis0 commented on how-to/install/vanilla.md:
```diff
@@ -0,0 +1,197 @@
+{% tab title="macOS 14 Sonoma or older" %}
+1. Download the [FO Installer](https://download.fo/vanilla)
+2. Open the `dmg` installer. If you get a security prompt from macOS, allow the installer to run:
+   * [Why did I get this prompt?](#why-does-macos-show-the-security-prompt)
```
see comment above

Madis0 commented on how-to/install/vanilla.md:
```diff
@@ -0,0 +1,197 @@
+1. On macOS and Linux, download and install [Java](https://download.fo/java)
+2. Download the [Fabric Installer](https://fabricmc.net/use)
+3. Open the Fabric Installer
+4. Select `version 0.16.9` of the Fabric Loader
```
```suggestion
4. Select `version 0.16.10` of the Fabric Loader
```

its-miroma commented on how-to/install/README.md:
```diff
@@ -0,0 +1,20 @@
+
+# Install
+
+<table data-card-size="large" data-view="cards"><thead><tr><th align="center"></th><th data-hidden data-card-target data-type="content-ref"></th></tr></thead><tbody><tr><td align="center"><strong>Modrinth App</strong><br>The Modrinth App supports FO with monetary benefits. Thank you!</td><td><a href="modrinth-app.md">modrinth-app.md</a></td></tr><tr><td align="center"><strong>CurseForge App</stro ...
```
all links are broken until all PRs are merged, that's why all new pages are hidden for now

its-miroma replied:
by the way, GitBook now has a broken links feature that warns if internal links are broken, so no worries!

its-miroma commented on crowdin.yml:
```diff
@@ -1,3 +0,0 @@
-files:
```
listing descriptions no longer exist, this repo should be disconnected from crowdin

Madis0 replied:
Good point, just seemed unexpected as part of this PR. Keep it in the PR.

its-miroma commented on .github/pull_request_template.md:
```diff
@@ -1,5 +0,0 @@
-<!--
```
the template mentions translations being not accepted through PRs, which is no longer relevant as there are no translations at all

Madis0 replied:
Good point, just seemed unexpected as part of this PR. Keep it in the PR.

its-miroma committed:
rewrite(how-to/install): fix

- fix urls
- fix Fabric version number
- change order of bullet points

its-miroma committed:
rewrite(how-to/install/multimc.md): fix urls

Madis0 commented on how-to/install/modrinth-app.md:
```diff
@@ -0,0 +1,22 @@
+
+{% hint style="success" %}
+The Modrinth App supports FO with monetary benefits. Thank you!
+{% endhint %}
```
Cache problem needs instructions, see https://wiki.download.fo/update-instructions#not-seeing-a-version-or-getting-an-error

its-miroma replied:
I had left a TODO in #125. Is it needed in both places?

Madis0 replied:
no, my bad commenting on the wrong file

Madis0 commented on how-to/install/multimc.md:
```diff
@@ -0,0 +1,67 @@
+
+{% tabs %}
+{% tab title="Automatic Updates (recommended)" %}
+1. Download and install [Java 21](https://download.fo/java21)
```
I suggest adding an info box above saying "While this variant allows for automatic updates within a specific Minecraft version, it is more difficult to update across Minecraft versions and to disable built-in mods if necessary. Consider using Prism Launcher instead, which also has a version updater."

Madis0 commented on how-to/install/multimc.md:
```diff
@@ -0,0 +1,67 @@
+{% endtab %}
+
+{% tab title="Easier Installation" %}
+1. Download and install [Java 21](https://download.fo/java21)
```
I suggest adding an info box above saying "Note that modpack updates require manual copying. Consider using Prism Launcher for a similar interface with an added version manager."

Madis0 commented on how-to/install/multimc.md:
```diff
@@ -0,0 +1,67 @@
+
+MultiMC does not natively support automatic updates of packs: see [issue #2640](https://github.com/MultiMC/MultiMC5/issues/2640) and [issue #3037](https://github.com/MultiMC/MultiMC5/issues/3057). FO chose to use `packwiz` because it is also used for the [vanilla installer](vanilla.md).
+
+Before FO 6.0.0, the MultiMC auto-updating pack was distributed on CurseForge, but that has changed for the following reasons:
```
This variant was never on CurseForge, the original MultiMC pack was. Though I don't think it is even necessary to mention it here.

Madis0 commented on how-to/install/multimc.md:
```diff
@@ -0,0 +1,67 @@
+
+{% hint style="info" %}
+You may also want to use one of the other [launchers supported by FO](./), which offer easier installation, updates and mod management.
+{% endhint %}
```
I suggest removing this to have more targeted messages below

its-miroma committed:
rewrite(how-to/install): update to latest versions

its-miroma committed:
rewrite(how-to/install/modrinth-app): via website

- use `kbd` for all UI buttons
- targeted hint for MultiMC
- use buttons for MultiMC
- drop inaccurate historical details about CurseForge MultiMC pack

its-miroma committed:
rewrite(how-to/install): sync with upstream

Madis0 commented:
Implemented via #145. Thank you!
